### PR TITLE
[Backend] Improve parameter patching

### PR DESCRIPTION
**Pull request by numerology**
Now it patches for all pipelines including preload and custom ones and tmp ones (e.g. ones created from `create_run_from_pipeline_func`. Also change XGboost and TFX sample to comply to new convention.

We now commit to the following conventional placeholders:
- `{{kfp-default-bucket}}` for default bucket deployed with KFP; and
- `{{kfp-project-id}}` for current GCP project ID.

Verified at: https://3ed47013582d1554-dot-us-central2.pipelines.googleusercontent.com/#/runs/details/876b4f92-b1dc-4b3a-9332-1e3a5d6ae1c8

Small side change: default remote build to false, otherwise we are always using remote building.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3016)

### CODE REVIEW

File: backend/src/apiserver/server/api_converter.go [hunk 1]
```diff
@@ -134,6 +141,14 @@ func toApiParameters(paramsString string) ([]*api.Parameter, error) {
 		var value string
 		if param.Value != nil {
 			value = *param.Value
+			// Patch the GCS default values if available
+			if common.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
+				var err error
+				value, err = PatchPipelineDefaultParameter(value)
+				if err != nil {
+					return nil, fmt.Errorf("failed to patch default value to pipeline. Error: %v", err)
```
Comment (neuromage) [inline backend/src/apiserver/server/api_converter.go]: Maybe return NewInternalServerError to be consistent?

Comment (numerology) [reply backend/src/apiserver/server/api_converter.go]: Sync with Yang, we decide to remove this part.

The actual replacement occurs in CreateRun. This part is only for informational purpose (user can get the actual parameter value from create run UI).

**Pull request merged**